### ISSUE TITLE
[BUG] Build C++ version but "metal::is_available()" is false.

### ISSUE THREAD

**Issue by HadXu** [ml-explore/mlx#731]
I build C++ version from scratch , and my Mac is support metal.

```python
Python 3.10.11 (main, Apr 20 2023, 13:58:42) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import mlx.core
>>> mlx.core.metal.is_available()
True
```

But when I create a new project include mlx

like

```bash
cmake_minimum_required(VERSION 3.27)
project(my_mlx LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 17)

include(FetchContent)
set(MLX_BUILD_METAL ON)

FetchContent_Declare(
        mlx
        GIT_REPOSITORY "https://github.com/ml-explore/mlx.git"
        GIT_TAG v0.3.0)
FetchContent_MakeAvailable(mlx)

add_executable(my_mlx main.cpp)
target_link_libraries(my_mlx PRIVATE mlx)
```

and the main.cpp is like

```cpp
#include <iostream>
#include <cassert>
#include "mlx/mlx.h"

#include "mlx/backend/metal/device.h"
#include "mlx/backend/metal/metal.h"

using namespace mlx::core;


int main() {
    std::cout << "Hello, World!" << std::endl;

    assert(metal::is_available());
    auto& device = metal::device(Device::gpu);

    return 0;
}
```
the assert is fail.

I was confused by this, is it a bug? or something should be set?

**Comment by HadXu** [comment]
I print the value, and the value is ```0```

```python
#include <iostream>
#include <cassert>
#include "mlx/mlx.h"

#include "mlx/backend/metal/device.h"
#include "mlx/backend/metal/metal.h"

using namespace mlx::core;


int main() {
    auto d = metal::is_available();
    auto& device = metal::device(Device::gpu);
    std::cout << d << std::endl;
    return 0;
}
```
the output is
```bash
0
```

**Comment by cemlyn007** [comment]
I am seeing this issue as well, am attempting to identify the root cause but I'm not a C++ expert!

**Comment by awni** [comment]
There is a compiler flag used to make sure that value gets set correctly. It looks like it isn't getting picked up properly when using MLX as a dep via Cmake. FWIW the code should run fine otherwise, but will definitely fix this.

**Comment by cemlyn007** [comment]
Maybe I don't understand CMake, but the branch is clearly going down the one that prints building metal source which means it should be running `add_compile_definitions(_METAL_)` and yet when I run your little main file I also see 0.
<img width="1512" alt="image" src="https://github.com/ml-explore/mlx/assets/49565105/a2ce929f-546d-467d-ba25-7f1e003ba0b1">

**Comment by awni** [comment]
Yes the problem is that ocmpile definition is used in the include itself so it has to be set by whomever uses the include. It's fixed in #739

**Comment by awni** [comment]
This is fixed as of #739
